hyperbahn/test/time-series/_time-series-cluster.js

```diff
@@ -0,0 +1,720 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var tapeCluster = require('tape-cluster');
+var setTimeout = require('timers').setTimeout;
+var parallel = require('run-parallel');
+var tape = require('tape');
+var nodeAssert = require('assert');
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+var process = require('process');
+var metrics = require('metrics');
+var console = require('console');
+var path = require('path');
+
+var HyperbahnCluster = require('../lib/test-cluster.js');
+
+var startOfFile = Date.now();
+
+TimeSeriesCluster.test = tapeCluster(tape, TimeSeriesCluster);
+
+TimeSeriesCluster.buildString = function buildString(size) {
+    var tempArray = [];
+
+    for (var i = 0; i < size; i++) {
+        tempArray.push('A');
+    }
+
+    return tempArray.join('');
+};
+
+module.exports = TimeSeriesCluster;
+
+function TimeSeriesCluster(opts) {
+    /* eslint max-statements: [2, 40] */
+    if (!(this instanceof TimeSeriesCluster)) {
+        return new TimeSeriesCluster(opts);
+    }
+
+    var self = this;
+
+    nodeAssert(opts && opts.buckets, 'requires buckets');
+    nodeAssert(opts && opts.clientTimeout, 'requires clientTimeout');
+    nodeAssert(opts && opts.endpointToRequest, 'requires endpointToRequest');
+    self.buckets = opts.buckets;
+    self.clientTimeout = opts.clientTimeout;
+    self.endpointToRequest = opts.endpointToRequest;
+
+    self.clientBatchDelay = opts.clientBatchDelay || 100;
+    self.numBatchesInBucket = opts.numBatchesInBucket || 5;
+    self.clientRequestsPerBatch = opts.clientRequestsPerBatch || 15;
+    self.serverInstances = opts.serverInstances || 10;
+    self.clientInstances = opts.clientInstances || 5;
+    self.requestBody = opts.requestBody || '';
+    self.requestBodyChunks = opts.requestBodyChunks || [];
```

Comment:
Please describe the semantics of each of these options in single-sentence comments.